src/list/List.tsx

```diff
@@ -46,6 +47,8 @@ function List<Item extends any>({
 
         if (listItemKeyGenerator) {
           key = listItemKeyGenerator(item, listItemTestId);
```

Comment:
I think we should re-arrange the ifs:
```ts
   if (listItemKeyGenerator) {
          key = listItemKeyGenerator(item, listItemTestId);
          // @ts-ignore
    }  else if (item && typeof item === "object" && item.id) {
        // @ts-ignore
        key = item.id;
        }  else {
          key = uuidv4();
        }
```
Because I believe currently we don't use id in any case. We override the first `if` in 43 in the `else` in 50.
Also I think it's time to create a util for this and clean up the component function body:
```tsx
        return <Fragment key={generateListItemKey({
          item, listItemTestId, listItemKeyGenerator
          })}>
          {children(item, listItemTestId, index)}
        </Fragment>;
```

Reply:
Updated it, thank you!